src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +441,50 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine },StringSplitOptions.None);
+    foreach (var headerElement in headerElements)
+    {
+        // Trim header for empty spaces
+        var header = headerElement.Trim();
+
+        // A header string must have atleast one semi-colon
+        if (GetSemiColons(header) > 0)

```

Comment:
It would seem like we don't care about the number of ~semi~colons in the header, rather we only care whether there is a ~semi~colon in the header, if so we could just do ```if (header.Contains(':'))``` and then drop the ```GetSemiColons()``` method.

Reply:
Fixed